jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/actions/CreateDataSourceActionTest.kt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.redshift.actions
+
+import com.intellij.database.autoconfig.DataSourceRegistry
+import com.intellij.testFramework.ProjectRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.redshift.model.Cluster
+import software.aws.toolkits.core.utils.RuleUtils
+import software.aws.toolkits.jetbrains.core.credentials.MockCredentialsManager
+import software.aws.toolkits.jetbrains.core.datagrip.CREDENTIAL_ID_PROPERTY
+import software.aws.toolkits.jetbrains.core.datagrip.REGION_ID_PROPERTY
+import software.aws.toolkits.jetbrains.core.region.MockRegionProvider
+import software.aws.toolkits.jetbrains.services.redshift.auth.IamAuth
+import software.aws.toolkits.jetbrains.services.redshift.createDatasource
+
+class CreateDataSourceActionTest {
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    @Test
+    fun `Add data source`() {
+        val port = 999
+        val address = RuleUtils.randomName()
+        val username = RuleUtils.randomName()
+        val registry = DataSourceRegistry(projectRule.project)
+        registry.createDatasource(projectRule.project,
+            Cluster.builder()
+                .endpoint { it.address(address).port(port) }
+                .masterUsername(username)
+                .build()
+        )
+        assertThat(registry.newDataSources).hasOnlyOneElementSatisfying {
+            assertThat(it.isTemporary).isFalse()
+            assertThat(it.url).contains(port.toString())
+            assertThat(it.url).contains(address)
+            assertThat(it.url).startsWith("jdbc:redshift://")
```

Comment:
why not check that the url completely matches?

Reply:
True, will change (once my IDE finishes indexing)